var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -60,6 +60,10 @@ class Openblas(MakefilePackage):
     patch('openblas_icc_openmp.patch', when='%intel@16.0:')
     patch('openblas_icc_fortran.patch', when='%intel@16.0:')
 
+    # Fixes compilation error on POWER8 with GCC 7
+    # https://github.com/xianyi/OpenBLAS/pull/1098
+    patch('power8.patch', when='@0.2.18:0.2.19')
```

Comment:
Just my opinion: for a patch as big as this one I think we should consider adding a `conflict` on `ppc64` (or whatever is the right platform) instead of patching the library. This is just to maintain the claim that we are working on *pristine sources*.

In any case (i.e. if people disagree with the above), I think patching should be done only when we are building on `power8` and with `gcc@7`:
```python
patch('power8.patch', when='@0.2.18:0.2.19 %gcc@7.1.0: target=ppc64')
```